File: MiniDOMFinanziaria/Finanziaria/CTEGFunEvaluator.vb
Imports minidom.Databases
Imports minidom.Sistema

Partial Public Class Finanziaria

    Public Class CTEGFunEvaluator
        Inherits FunEvaluator

        Public Durata As Integer
        Public Quota As Decimal
        Public NettoRicavo As Decimal
        Public SpeseAssicurative As Decimal
        Public Imposte As Decimal

        Public Overrides Function EvalFunction(x As Double) As Double
            Dim s As Integer
            Dim ret As Double = 0
            For s = 1 To Me.Durata Step 1
                ret = ret + (CDbl(1) + x) ^ (-s / CDbl(12))
            Next
            ret = CDbl(ret) * Quota - (NettoRicavo + SpeseAssicurative + Imposte)
            Return ret
        End Function

    End Class


End Class

File: MiniDOMFinanziaria/Finanziaria/CTEGFunEvaluator.cs

namespace minidom
{
    public partial class Finanziaria
    {
        public class CTEGFunEvaluator : FunEvaluator
        {
            public int Durata;
            public decimal Quota;
            public decimal NettoRicavo;
            public decimal SpeseAssicurative;
            public decimal Imposte;

            public override double EvalFunction(double x)
            {
                int s;
                double ret = 0d;
                var loopTo = Durata;
                for (s = 1; s <= loopTo; s += 1)
                    ret = ret + Maths.Pow(1 + x, (double)-s / 12);
                ret = ret * (double)Quota - (double)(NettoRicavo + SpeseAssicurative + Imposte);
                return ret;
            }
        }
    }
}